libraries/gpu-gl-common/src/gpu/gl/GLTextureTransfer.cpp

```diff
@@ -218,7 +236,19 @@ void GLTextureTransferEngineDefault::updateMemoryPressure() {
 
     Backend::textureResourceIdealGPUMemSize.set(idealMemoryAllocation);
     size_t unallocated = idealMemoryAllocation - totalVariableMemoryAllocation;
-    float pressure = (float)totalVariableMemoryAllocation / (float)allowedMemoryAllocation;
+    float pressure = 0;
+
+    if (useAvailableGlMemory) {
+        float totalMem = GLBackend::getTotalMemory();
+        float availMem = GLBackend::getAvailableMemory() - AUTO_RESERVE_TEXTURE_MEMORY;
+        if (availMem < 0) {
+            availMem = 0;
+        }
+
+        pressure = (totalMem - availMem) / totalMem;
+    } else {
+        pressure = (float)totalVariableMemoryAllocation / (float)allowedMemoryAllocation;
```

Comment:
can either of the denominators (totalMem or allowedMemoryAllocation) be 0?  should we detect that and do something else?  otherwise looks good

Reply:
allowedMemoryAllocation is dealt with early on starting on line 191. If zero (this signals an "auto" setting), it sets useAvailableGlMemory only if the backend says we know how much free VRAM we have, and in that case there is no fixed allowed allocation, we just keep watch on free memory. totalMem is guaranteed to be known, because if we know how much is free, we definitely know the total.

If free memory is not known (eg, Intel), then we pick 80% of the maximum as a target (idea being this is an Intel card, and so probably nothing else is running. This is just a guess, but I don't see any better options).

If nothing is known, fallback to the old system of assuming a 1 GB limit.